vCardLib/Deserialization/FieldDeserializers/EmailAddressFieldDeserializer.cs

```diff
@@ -23,20 +23,26 @@ public EmailAddress Read(string input)
         EmailAddressType? type = null;
         int? preference = null;
 
-        foreach (var datum in metadata)
+        foreach (var datum in metadata) 
         {
             var (key, data) = DataSplitHelpers.SplitDatum(datum, '=');
 
-            if (key.EqualsIgnoreCase(FieldKeyConstants.TypeKey) && data != null)
+            if (key.EqualsIgnoreCase(FieldKeyConstants.PreferenceKey)) 
             {
-                var emailTypes = data.Split(FieldKeyConstants.ConcatenationDelimiter);
-
-                type = emailTypes
-                    .Select(parsedType => parsedType.ParseEmailAddressType())
-                    .Aggregate(type, (current, emailType) => current.HasValue ? current.Value | emailType : emailType);
-            }
-            else if (key.EqualsIgnoreCase(FieldKeyConstants.PreferenceKey))
                 preference = 1;
+                continue;
+            }
+
+            var emailTypes = key.EqualsIgnoreCase(FieldKeyConstants.TypeKey) && data != null
+                ? data.Split(',')
+                : key != null && data == null
+                    ? key.Split(';')
+                    : null;
+
+            type = emailTypes?
+                .Select(parsedType => parsedType.ParseEmailAddressType())
+                .Where(parsedType => parsedType != null)
+                .Aggregate(type, (current, emailType) => current.HasValue ? current.Value | emailType : emailType);
```

Comment:
Hi @ManolisKoufidakis I think this can be simplified using the helper methods

```csharp
var (key, data) = DataSplitHelpers.ExtractKeyValue(metadatum, '=');

  if (key is null || key.EqualsIgnoreCase(FieldKeyConstants.TypeKey))
  {
      var parsedType = data.ParseEmailAddressType();
      type = type is null ? parsedType : type | parsedType;
  }
  else if (key.EqualsIgnoreCase(FieldKeyConstants.PreferenceKey))
                  pref = int.Parse(data);
```